src/relentless/simulate/hoomd.py

```diff
@@ -145,26 +149,62 @@ def _get_masses_from_snapshot(self, sim, snap):
         masses.update(masses_)
         return masses
 
+    def _assert_dimension_safe(self, sim, snap):
+        if sim.dimension == 3:
+            dim_safe = True
+        elif sim.dimension == 2:
+            if mpi.world.rank == 0:
+                dim_safe = numpy.allclose(
+                    snap.particles.position[:, 2], 0
+                ) and numpy.allclose(snap.particles.velocity[:, 2], 0)
+            else:
+                dim_safe = None
+            dim_safe = mpi.world.bcast(dim_safe, root=0)
+        else:
+            raise ValueError("Only 2d and 3d simulations are supported")
 
-class InitializeFromFile(InitializationOperation):
-    """Initialize a simulation from a GSD file.
-
-    Parameters
-    ----------
-    filename : str
-        The file from which to read the system data.
+        if not dim_safe:
+            raise ValueError("Simulation initialized inconsistent with dimension")
 
-    """
 
-    def __init__(self, filename):
-        self.filename = os.path.realpath(filename)
+class InitializeFromFile(InitializationOperation):
+    def __init__(self, filename, format, dimension):
+        self.filename = filename
+        self.format = format
+        self.dimension = dimension
 
     def _initialize_v3(self, sim):
-        sim["engine"]["_hoomd"].create_state_from_gsd(self.filename)
+        gsd_filename = self._convert_to_gsd_file(sim)
+        sim["engine"]["_hoomd"].create_state_from_gsd(gsd_filename)
 
     def _initialize_v2(self, sim):
+        gsd_filename = self._convert_to_gsd_file(sim)
         with sim["engine"]["_hoomd"]:
-            return hoomd.init.read_gsd(self.filename)
+            return hoomd.init.read_gsd(gsd_filename)
+
+    def _convert_to_gsd_file(self, sim):
+        file_format = initialize.InitializeFromFile._detect_format(
+            self.filename, self.format
```

Comment:
Does this have the same issue with not wrapping this `sim.directory.file(self.filename)`? Maybe we should add a line at the top:
```py
filename = sim.directory.file(self.filename)
```
Then use that rather than `self.filename` everywhere?

Reply:
I don't think these are an issue. The `_detect_format()` function is just checking the filename as a string. The location of the file doesn't impact its ability to do that check

Reply:
What about down below where it reads the `DataFile` though? (That was actually what I was looking at.) I thought it might just be more readable in general to use the _same_ filename throughout the function.

Reply:
I'm not sure why it didn't cause trouble with the `DataFile` below. I agree that it would be better to use the same filename throughout!